pkg/controller/node/node_controller.go

```diff
@@ -265,10 +265,21 @@ func (ctrl *Controller) getPoolForNode(node *corev1.Node) (*mcfgv1.MachineConfig
 
 	var pools []*mcfgv1.MachineConfigPool
 	for _, p := range pl {
+		// first get MachineSelector
 		selector, err := metav1.LabelSelectorAsSelector(p.Spec.MachineSelector)
 		if err != nil {
 			return nil, fmt.Errorf("invalid label selector: %v", err)
 		}
+		// if MachineSelector is empty, get and use NodeSelector
+		if selector == nil {
+			selector, err = metav1.LabelSelectorAsSelector(p.Spec.NodeSelector)
+			if err != nil {
+				return nil, fmt.Errorf("invalid label selector: %v", err)
+			}
+		} else {
```

Comment:
wait hmm do i need this else?

Reply:
nope, as it's gonna be the initial machineSelector you grabbed at line 268/269

Reply:
there are other spots that we need this 

https://github.com/openshift/machine-config-operator/blob/76232a00900850733d05b049d82683cbdb1401e0/pkg/controller/node/node_controller.go#L438

https://github.com/openshift/machine-config-operator/blob/76232a00900850733d05b049d82683cbdb1401e0/pkg/controller/node/node_controller.go#L450

Reply:
gotcha!